src/lava/proc/dense/models.py

```diff
@@ -31,14 +31,19 @@ class PyDenseModelFloat(PyLoihiProcessModel):
     weight_exp: float = LavaPyType(float, float)
     num_weight_bits: float = LavaPyType(float, float)
     sign_mode: float = LavaPyType(float, float)
+    use_graded_spike: np.ndarray = LavaPyType(np.ndarray, bool, precision=1)
 
     def run_spk(self):
         # The a_out sent on a each timestep is a buffered value from dendritic
         # accumulation at timestep t-1. This prevents deadlocking in
         # networks with recurrent connectivity structures.
         self.a_out.send(self.a_buff)
-        s_in = self.s_in.recv()
-        self.a_buff = self.weights[:, s_in].sum(axis=1)
+        if self.use_graded_spike.item():
+            s_in = self.s_in.recv()
+            self.a_buff = self.weights.dot(s_in)
+        else:
+            s_in = self.s_in.recv().astype(bool)
+            self.a_buff = self.weights[:, s_in].sum(axis=1)
```

Comment:
Same here. I'm in favor or self.weights.dot(s_in) for all spike types, unless we can prove the boolean summing is significantly faster. I suspect with numpy optimization they're not that different?

Reply:
ibid.